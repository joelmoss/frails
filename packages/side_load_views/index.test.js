const serializer = require('jest-serializer-path')
const { createConfig } = require('@frails/core')

const sideLoadViews = require('.')

describe('sideLoadViews', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('createConfig', () => {
    expect(createConfig(sideLoadViews)).toMatchSnapshot()
    expect(JSON.stringify(createConfig(sideLoadViews()))).toBe(
      JSON.stringify(createConfig(sideLoadViews))
    )
  })

  test('disabled', () => {
    expect(createConfig(sideLoadViews(false))).toMatchSnapshot()
  })
})
