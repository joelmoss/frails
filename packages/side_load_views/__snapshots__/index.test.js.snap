// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sideLoadViews createConfig 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {
    "views/layouts/application": Array [
      "<PROJECT_ROOT>/app/views/layouts/application.css",
      "<PROJECT_ROOT>/app/views/layouts/application.js",
    ],
    "views/layouts/no_side": Array [
      "<PROJECT_ROOT>/app/views/layouts/no_side.css",
    ],
    "views/pages/_header": Array [
      "<PROJECT_ROOT>/app/views/pages/_header.css",
      "<PROJECT_ROOT>/app/views/pages/_header.js",
    ],
    "views/pages/_post": Array [
      "<PROJECT_ROOT>/app/views/pages/_post.css",
    ],
    "views/pages/home": Array [
      "<PROJECT_ROOT>/app/views/pages/home.css",
    ],
  },
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;

exports[`sideLoadViews disabled 1`] = `
Object {
  "context": "<PROJECT_ROOT>/app",
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "modules": Array [
      "<PROJECT_ROOT>/app",
    ],
  },
}
`;
